### PR TITLE
Refactor to camerax / boofCV, POC (EXPOSUREAPP-11089)

### PR DESCRIPTION
**Pull request by BMItr**
This PR addresses EXPOSUREAPP-11089.

**Contains:** 
Refactored UQS Scanner-Fragment and related classes.
Refactored Submission On-behalf Scanner Fragment and related classes.
Use of newest version of CameraX/boofCV.

**Note:**
POC for testing!

**How to test:**
- Open the Universal QR Code Scanner inside the app
- Scan some certificates
- Check if everything works fine

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/CameraHelper.kt [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        } catch (e: Exception) {
+            Timber.tag(TAG).e(e)
+        }
+    }
+
+    private fun onImage(imageProxy: ImageProxy) {
+        if (scanEnabled) {
+            onImageCallback(imageProxy)
+        } else {
+            imageProxy.use { }
```
[the remark this answers is not in the document]
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/CameraHelper.kt:121]: Simply closing the image in a safe way

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/CameraHelper.kt [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                }
+            }
+
+        try {
+            val cameraProvider = cameraProvider ?: throw IllegalStateException("Camera not initialized")
+            cameraProvider.unbindAll()
+
+            camera = cameraProvider.bindToLifecycle(lifecycleOwner, selector, preview, analyzer)
+        } catch (e: Exception) {
+            Timber.tag(TAG).e(e)
```
[the remark this answers is not in the document]
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/CameraHelper.kt:113]: Yes, but only if the PoC proves to be worth it

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/CameraHelper.kt [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import androidx.lifecycle.LifecycleOwner
+import androidx.window.WindowManager
+import de.rki.coronawarnapp.tag
+import timber.log.Timber
+import java.util.concurrent.Executors
+import kotlin.math.abs
+import kotlin.math.max
+import kotlin.math.min
+
+class CameraHelper(
```
[the remark this answers is not in the document]
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/CameraHelper.kt:22]: PreviewView is final, so it cannot be extended. But I will create a custom view with an PreviewView in it

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/presencetracing/organizer/warn/qrcode/OrganizerWarnQrCodeScannerFragment.kt [hunk 4]
```diff
@@ -85,33 +85,22 @@ class OrganizerWarnQrCodeScannerFragment : Fragment(R.layout.fragment_qrcode_sca
                 }
                 is OrganizerWarnQrCodeNavigation.Error ->
                     navEvent.exception.toErrorDialogBuilder(requireContext()).show()
-                OrganizerWarnQrCodeNavigation.InProgress -> binding.qrCodeProcessingView.isVisible = true
+                OrganizerWarnQrCodeNavigation.InProgress,
+                OrganizerWarnQrCodeNavigation.Scanning -> {
+                }
```
[the remark this answers is not in the document]
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/presencetracing/organizer/warn/qrcode/OrganizerWarnQrCodeScannerFragment.kt:90]: Will add comments

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/QrCodeScannerFragment.kt [hunk 5]
```diff
@@ -111,11 +112,12 @@ class QrCodeScannerFragment : Fragment(R.layout.fragment_qrcode_scanner), AutoIn
...
                     else -> showScannerResultErrorDialog(scannerResult.error)
                 }
-
-                InProgress -> binding.qrCodeProcessingView.isVisible = true
                 InfoScreen -> doNavigate(
                     QrCodeScannerFragmentDirections.actionUniversalScannerToUniversalScannerInformationFragment()
                 )
+                InProgress,
+                Scanning -> {
+                }
```
[the remark this answers is not in the document]
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/QrCodeScannerFragment.kt:120]: Will add comments

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/CameraHelper.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/CameraHelper.kt]: done :-)

**Review by mtwalli — APPROVED** [review]

**Pull request merged**